### DISCUSSION TITLE
Dynamically assign property values outside of an addrmap scope?

### DISCUSSION THREAD

**Discussion by greedysquid** [SystemRDL/systemrdl-compiler#97] [Q&A] [answered]
Is is possible to dynamically assign property values outside of an addrmap scope?
The below code doesn't compile, but hopefully shows what am trying to achieve...
```
addrmap my_addrmap {
    reg my_reg {
        field {
            reset=0;
        } my_field;
    };
};
my_addrmap.my_reg.my_field->reset=1;
```
[My aim is to have a generic addrmap that I can then modify outside of the addrmap, to create variants of that addmap for regression testing.]
Thanks

**Comment by amykyta3** [comment, marked as answer]
That wont work since `my_addrmap` is not an instance. However, you could use RDL parameters:
```
addrmap my_addrmap #(longint MY_RESET = 0) {
    reg my_reg {
        field {
            reset = MY_RESET;
        } my_field;
    };
};
```

You can override the parameter's value during elaboration: https://systemrdl-compiler.readthedocs.io/en/latest/api/compiler.html#systemrdl.RDLCompiler.elaborate
For example:
```python
rdlc = RDLCompiler()
rdlc.compile_file('my_file.rdl')
root = rdlc.elaborate(parameters={'MY_RESET': 1})
```

**Reply by greedysquid** [reply]
Thanks for the (prompt) answer.
Marking as the answer - because you did answer my question.
Have chosen not to use RDL parameters as my application may/will require many reset values to be optionally overridden, so using RDL parameters would get hard to manage quite quickly.
However, your comment about `my_addrmap` not being an instance did give me another idea - embedding `my_addrmap` within a parent addrmap, so that `my_addrmap` has to be created as an instance.
(Although this didn't work out for me because of my particular file structure - register definitions in different files that are included in the main file.)
So I ended up compiling `my_addrmap` and then changing the reset values of the python objects directly.
